spec/helpers/claims_helper_spec.rb

```diff
@@ -212,4 +212,72 @@ def current_user
       it { is_expected. to be_falsey }
     end
   end
+
+  describe '#messaging_permitted?' do
+    subject { messaging_permitted?(message) }
+
+    let(:claim) { build :claim, state: state }
+    let(:message) { build(:message, claim: claim, claim_action: claim_action) }
+    let(:claim_action) { nil }
+
+    helper do
+      def current_user
+        instance_double(User, persona: persona)
+      end
+    end
+
+    context 'for case_worker' do
+      let(:persona) { create :case_worker }
+
+      context 'for a claim with claim actions' do
+        let(:state) { 'rejected' }
+        let(:claim_action) { 'Request written reasons' }
+
+        it { is_expected. to be_truthy }
+      end
+
+      %w[submitted allocated authorised part_authorised rejected refused redetermination awaiting_written_reasons].each do |state|
+        context "when claim state is #{state}" do
+          let(:state) { state }
+
+          it { is_expected. to be_falsey }
+        end
+      end
+    end
+
+    context 'for external_user' do
+      let(:persona) { create :external_user }
+
+      context 'for a claim with claim actions' do
+        let(:state) { 'rejected' }
+        let(:claim_action) { 'Request written reasons' }
+
+        it { is_expected. to be_truthy }
+      end
+
+      context 'for non redeterminable claim states' do
+        let(:claim) { build :claim, state: state }
+
+        %w[submitted allocated redetermination awaiting_written_reasons].each do |state|
+          context "when claim state is #{state}" do
+            let(:state) { state }
+
+            it { is_expected. to be_truthy }
+          end
+        end
+      end
+
+      context 'for redeterminable claim states' do
+        let(:claim) { build :claim, state: state }
+
+        %w[authorised part_authorised rejected refused].each do |state|
+          context "when claim state is #{state}" do
+            let(:state) { state }
+
+            it { is_expected. to be_falsey }
```

Comment:
does this not mean that messaging will be disabled for any claim in one of these states?

Reply:
The BA tested this so and said the functionality is as expected for non iterims too, so am going to merge